### PR TITLE
add iotaSlice variant with step

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: std/experimental/ndslice/selection.d [hunk 1]
```diff
@@ -1770,6 +1779,15 @@ IotaSlice!N iotaSlice(size_t N)(auto ref size_t[N] lengths, size_t shift = 0)
     assert(cm[2, 1] == 119); // 119 = 100 + (1 + 2) * 5 + (3 + 1)
 }
 
+/// `iotaSlice` with step
+@safe pure nothrow unittest
```
Comment (wilzbach) [inline std/experimental/ndslice/selection.d]: Do you want to make that test `@nogc` too?

Comment (9il) [inline std/experimental/ndslice/selection.d]: no need, without `@nogc` it looks better

File: std/experimental/ndslice/selection.d [hunk 2]
```diff
@@ -1742,6 +1742,15 @@ IotaSlice!N iotaSlice(size_t N)(auto ref size_t[N] lengths, size_t shift = 0)
     with (typeof(return)) return Range.init.sliced(lengths, shift);
 }
 
+///ditto
+IotaSlice!N iotaSlice(size_t N)(auto ref size_t[N] lengths, size_t shift, size_t step)
+{
+    auto iota = iotaSlice(lengths, shift);
+    foreach (i; Iota!(0, N))
+        iota._strides[i] *= step;
```
Comment (wilzbach) [inline std/experimental/ndslice/selection.d]: I have to say that's pretty clever. Nice :)

**Comment by wilzbach** [comment]
LGTM, very nice addition!

File: std/experimental/ndslice/selection.d [hunk 3]
```diff
@@ -1742,6 +1742,15 @@ IotaSlice!N iotaSlice(size_t N)(auto ref size_t[N] lengths, size_t shift = 0)
     with (typeof(return)) return Range.init.sliced(lengths, shift);
 }
 
+///ditto
```
Comment (ZombineDev) [inline std/experimental/ndslice/selection.d]: Please add `step` to the parameters docs.

**Comment by ZombineDev** [comment]
LGTM too, modulo doc nitpick

**Comment by 9il** [comment]
> LGTM too, modulo doc nitpick

thanks, `step` added )

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**